packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_control_pane.dart

```diff
@@ -15,24 +15,41 @@ class SnapshotControlPane extends StatelessWidget {
       : super(key: key);
 
   final DiffPaneController controller;
+  static const _classFilterWidth = 200.0;
 
   @override
   Widget build(BuildContext context) {
     return ValueListenableBuilder<bool>(
       valueListenable: controller.isProcessing,
       builder: (_, isProcessing, __) {
-        final current = controller.selectedItem as SnapshotListItem;
+        final current = controller.core.selectedItem as SnapshotInstanceItem;
 
         return Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
           children: [
-            const SizedBox(width: defaultSpacing),
-            if (!isProcessing && current.heap != null) ...[
-              _DiffDropdown(
-                current: current,
-                list: controller.snapshots,
-              ),
-              const SizedBox(width: defaultSpacing),
-            ],
+            Row(
+              children: [
+                const SizedBox(width: defaultSpacing),
+                if (!isProcessing && current.heap != null) ...[
+                  _DiffDropdown(
+                    current: current,
+                    controller: controller,
+                  ),
+                  const SizedBox(width: defaultSpacing),
+                  SizedBox(
+                    width: _classFilterWidth,
+                    child: _ClassFilter(onChanged: controller.setClassFilter),
+                  ),
+                  const SizedBox(width: defaultSpacing),
+                  ToCsvButton(
+                    minScreenWidthForTextBeforeScaling:
+                        primaryControlsMinVerboseWidth,
+                    onPressed: current.downloadToCsv,
+                  ),
+                ],
+              ],
+            ),
+            // This child is aligned to the right.
```

Comment:
remove comment. This is implicit by using `mainAxisAlignment: MainAxisAlignment.spaceBetween`